From a77cd07acfb145465e396c835090fe00bf713afe Mon Sep 17 00:00:00 2001
From: John Brooks <john.brooks@dereferenced.net>
Date: Mon, 29 Apr 2013 02:28:50 -0600
Subject: [PATCH] Disable parallel build for extensions directory

---
 extensions/Makefile.am |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/extensions/Makefile.am b/extensions/Makefile.am
index ecb02ef..fbd30e7 100644
--- a/extensions/Makefile.am
+++ b/extensions/Makefile.am
@@ -93,3 +93,6 @@ Android.mk: Makefile.am $(BUILT_SOURCES)
 	 -:CPPFLAGS $(CPPFLAGS) $(AM_CPPFLAGS) \
 	 -:LDFLAGS $(libgabble_extensions_la_LIBADD) \
 	> $@
+
+.NOTPARALLEL:
+	
-- 
1.7.9.5

