diff --git a/autogen.sh b/autogen.sh
index 0d8e27d..e9459e3 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -1,7 +1,27 @@
 #!/bin/sh
 set -e
 
-gtkdocize || exit 1
+# check if gtk-doc is explicitly disabled
+for ag_option in $@
+do
+  case $ag_option in
+    -disable-gtk-doc | --disable-gtk-doc)
+    enable_gtk_doc=no
+  ;;
+  esac
+done
+
+if test x$enable_gtk_doc = xno; then
+    if test -f gtk-doc.make; then :; else
+       echo "EXTRA_DIST = missing-gtk-doc" > gtk-doc.make
+    fi
+    echo "WARNING: You have disabled gtk-doc."
+    echo "         As a result, you will not be able to generate the API"
+    echo "         documentation and 'make dist' will not work."
+    echo
+else
+    gtkdocize || exit $?
+fi
 
 if test -n "$AUTOMAKE"; then
     : # don't override an explicit user request
diff --git a/configure.ac b/configure.ac
index 75c1c48..f89952e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -286,7 +286,13 @@ dnl To be used by tests and examples
 WOCKY_CFLAGS='-I${top_builddir} -I${top_srcdir}'
 AC_SUBST(WOCKY_CFLAGS)
 
-GTK_DOC_CHECK([1.17],[--flavour no-tmpl])
+AM_CONDITIONAL(ENABLE_GTK_DOC, test "x$enable_gtk_doc" = xyes)
+AM_CONDITIONAL(DISABLE_GTK_DOC, test "x$disable_gtk_doc" = xno)
+# Check for GTK_DOC_CHECK availability. The GTK_DOC_CHECK invocation
+# must be on its own line, gtkdocize relies on it
+m4_ifdef([GTK_DOC_CHECK], [
+ GTK_DOC_CHECK([1.17],[--flavour no-tmpl])
+])
 
 AC_OUTPUT( Makefile            \
            wocky/Makefile      \
